src/pages/groupComparison/GroupComparisonUtils.ts

```diff
@@ -8,4 +9,23 @@ export type SampleGroup = {
     legendText?: string;
 };
 
-export const TEMP_localStorageGroupsKey = "__tmp__groupComparisonGroups";
\ No newline at end of file
+export const TEMP_localStorageGroupsKey = "__tmp__groupComparisonGroups";
+
+export function getCombinations(sets: { name: string, cases: string[] }[]) {
+    let combinations: { sets: string[], cases: string[] }[] = [];
+
+    let f = function (res: { sets: string[], cases: string[] }, sets: { name: string, cases: string[] }[]) {
+        for (let i = 0; i < sets.length; i++) {
+            let currentSet = sets[i];
+            let commonCases = res.cases.length === 0 ? currentSet.cases : _.intersection(res.cases, currentSet.cases)
+            let newSet = {
+                sets: [...res.sets, currentSet.name],
+                cases: commonCases
+            }
+            combinations.push(newSet);
+            f(newSet, sets.slice(i + 1));
+        }
+    }
+    f({ sets: [], cases: [] }, sets);
+    return combinations;
```

Comment:
while I appreciate the recursive solution, I think we should simplify this by hardcoding the combinations for up to three groups, since that's all we need. To reflect this more in the code, the function parameter could also take a tuple type of two elements with an optional third `(sets: [SampleGroup, SampleGroup, SampleGroup?])` 

what do you think?

Reply:
also, why have you not taken SampleGroup[] as the argument here?

Reply:
there is a possibility to list all the combinations in a table. I would prefer to leave as is and may be send only 3 groups while invoking it. 
and `SampleGroup` has `SampleIdentifier` which doesn't have any information about patient

Reply:
ah, I see. sounds good

Reply:
I think it deserves a unit test